python/mp/rosary.py

```diff
@@ -231,28 +281,38 @@ def register_written_triggers(self):
     ##########################################################################
     # ROSARY CONTROL - RUNTIME HUMAN INTERFACES
     ##########################################################################
-    def update(self):
+    def update(self, bead_list):
         """Transmit the OSC message to update all beads on the rosary."""
-        bundle = osc_bundle_builder.OscBundleBuilder(osc_bundle_builder.IMMEDIATELY)
-
-        i = 0
-        while (i < self.BEAD_COUNT):
-            msg = osc_message_builder.OscMessageBuilder(address = "/beadf")
-            msg.add_arg(i)
-            msg.add_arg(float(self.beads[i].color.r))
-            msg.add_arg(float(self.beads[i].color.g))
-            msg.add_arg(float(self.beads[i].color.b))
-            msg = msg.build()
-            bundle.add_content(msg)
-            i = i + 1
-            
-        msg = osc_message_builder.OscMessageBuilder(address = "/update")
-        msg = msg.build()
-        bundle.add_content(msg)
+        #bundle = osc_bundle_builder.OscBundleBuilder(osc_bundle_builder.IMMEDIATELY)
+
+        i = int(0)
+
+        msg = osc_message_builder.OscMessageBuilder(address = "/bead/")
+        msg.add_arg(int(0))    # base
+        msg.add_arg(int(len(bead_list)))  # length
+
+        payload = bytearray()
+        for bead in bead_list:
+
+
+            payload.append((int(bead.color.r * 255)))
+            payload.append((int(bead.color.g * 255)))
+            payload.append((int(bead.color.b * 255)))
+
+        #print(bytes(payload))
+
+        #print('len(bead_list)', len(bead_list), 'len(payload)', len(payload))
+        
+        msg.add_arg(bytes(payload))
             
-        bundle = bundle.build()
-        self.osc_client.send(bundle)
+        # msg = osc_message_builder.OscMessageBuilder(address = "/update")
 
+        #bundle.add_content(msg)
+            
+        #bundle = bundle.build()
+        #self.osc_client.send(bundle)
+        self.osc_client.send(msg.build())
```

Comment:
If I understand the point of this PR correctly, it's now the job of the Updater classes to call update() and set the colors on the individual beads? What do the changes here do then? Or rather, what does rosary.update() do now?